lib/dao.js

```diff
@@ -1597,6 +1598,22 @@ DataAccessObject._coerce = function(where) {
 
       continue;
     }
+
+    if (p.match(/\./)) {
+      var model = p.split('.')[0];
+      var prop = p.split('.').slice(1);
+
+      if (props[model]) {
+        var clause = {};
+        clause[prop] = where[p];
+        where[p] = Array.isArray(props[model].type) ?
+          self._coerce(clause, props[model].type[0].definition.properties)[prop] :
+          self._coerce(clause, props[model].type.definition.properties)[prop];
```

Comment:
Can we rely on `...type.definition` to exist and have `.properties`? I think it would be safer to add a check.

```js
clause[prop = where[p];
var type = Array.isArray(props[model].type) ? props[model].type[0] : props[model].type;
if (type && type.definition && type.definition.properties) {
  where[p] = self._coerce(clause, type)[prop];
  continue;
}
// else fall-back to old/default coercion
```